tests/testthat/test-independent-ahr_blinded.R

```diff
@@ -0,0 +1,161 @@
+
+  # Test 3: Blinded AHR (theta) is computed correctly
+  theta <- -sum(log(hr[1:nx]) * event) / sum(event)
+  expect_true(!is.na(theta))
+
+  # Test 4: Information adjustment (q_e) is computed correctly
+  q_e <- ratio / (1 + ratio)
+  expect_equal(q_e, 2 / 3)
+
+  # Check the overall result
+  expect_true(inherits(result, "tbl_df"))
+  expect_equal(result$event, sum(event))
+  expect_equal(result$theta, theta)
+  expect_equal(result$ahr, exp(-theta))
+})
+
+test_that("Correct computation of blinded AHR and information adjustment", {
+  surv <- survival::Surv(simtrial::ex2_delayed_effect$month, event = simtrial::ex2_delayed_effect$evntd)
+  intervals <- c(3, Inf)
+  hr <- c(1, 0.6)
+  ratio <- 1
+  event <- simtrial::fit_pwexp(surv, intervals)[, 3]
+
+  expected_event <- sum(surv[, "status"])
+  expected_theta <- -sum(log(hr[1:length(event)]) * event) / sum(event)
+  expected_ahr <- exp(-(-sum(log(hr[1:length(event)]) * event) / sum(event)))
+  expected_info0 <- sum(surv[, "status"]) * (1 - ratio / (1 + ratio)) * (ratio / (1 + ratio))
+
+  result <- ahr_blinded(surv = surv, intervals = intervals, hr = hr, ratio = ratio)
+
+  expect_equal(result$event, expected_event)
+  expect_equal(result$ahr, expected_ahr, tolerance = 0.001)
+  expect_equal(result$theta, expected_theta, tolerance = 0.001)
+  expect_equal(result$info0, expected_info0)
+})
+
+test_that("ahr_blinded computes theta with constant hazard ratios correctly", {
+  surv <- survival::Surv(
+    simtrial::ex1_delayed_effect$month,
+    simtrial::ex1_delayed_effect$evntd
+  )
+  intervals <- c(3, 6, Inf)
+  hr <- c(1, 1, 1)
+  result <- ahr_blinded(surv = surv, intervals = intervals, hr = hr)
+
+  # When all hr = 1, theta should be 0
+  expect_equal(result$theta, 0)
+})
+
+test_that("ahr_blinded handles zero events", {
+  surv <- survival::Surv(time = c(1, 2, 3, 4, 5), event = c(0, 0, 0, 0, 0)) # No events
+  intervals <- c(2, 3, Inf)
+  hr <- c(0.8, 0.9, 1)
+
+  result <- ahr_blinded(surv = surv, intervals = intervals, hr = hr)
+
+  expect_equal(result$event, 0)
+  expect_true(is.nan(result$ahr))
+  expect_true(is.nan(result$theta))
+  expect_equal(result$info0, 0)
+})
+
+test_that("ahr_blinded handles all events in the first interval", {
+  surv <- survival::Surv(time = c(1, 2, 2.5, 3, 3.5), event = c(1, 1, 1, 1, 1))
+  # Only the first interval contains events
+  intervals <- c(4, Inf)
+  hr <- c(0.5, 0.7)
+
+  result <- ahr_blinded(surv = surv, intervals = intervals, hr = hr)
+
+  # All events are in the first interval, so result should reflect hr[1]
+  expected_theta <- -sum(log(hr[1]) * sum(surv[, "status"])) / sum(surv[, "status"])
+  expect_equal(result$theta, expected_theta)
+  expect_equal(result$event, sum(surv[, "status"]))
+})
+
+test_that("ahr_blinded handles very small or very large hazard ratios", {
+  surv <- survival::Surv(time = c(1, 2, 3, 4, 5), event = c(1, 1, 1, 1, 1))
+  intervals <- c(2, 3, Inf)
+  hr_small <- c(0.1, 0.2, 0.3)
+  hr_large <- c(10, 20, 30)
```

Comment:
HR is always a number between 0 and 1. So it is not statistically meaningful to have `hr_large = c(10, 20, 30)`.